### ISSUE TITLE
[spv-in] Naga rejects example with break from switch

### ISSUE THREAD

**Issue by afd** [gfx-rs/naga#1028]
Naga (1f42d4f2271adfa6a99427fde84cd2b13dbaa215) rejects this SPIR-V:
```
               OpCapability Shader
          %1 = OpExtInstImport "GLSL.std.450"
               OpMemoryModel Logical GLSL450
               OpEntryPoint Fragment %4 "main"
               OpExecutionMode %4 OriginUpperLeft
          %2 = OpTypeVoid
          %3 = OpTypeFunction %2
         %72 = OpTypeInt 32 0
         %73 = OpConstant %72 0
        %114 = OpTypeBool
       %2317 = OpUndef %114
          %4 = OpFunction %2 None %3
       %1538 = OpLabel
               OpSelectionMerge %1729 None
               OpSwitch %73 %1608
       %1608 = OpLabel
               OpSelectionMerge %1621 None
               OpBranchConditional %2317 %1619 %1621
       %1619 = OpLabel
               OpBranch %1729
       %1621 = OpLabel
               OpBranch %1729
       %1729 = OpLabel
               OpReturn
               OpFunctionEnd
```

I get:

```
$ RUST_BACKTRACE=1 cargo run ./reductiontemp/reduced.spv temp.wgsl
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/naga ./reductiontemp/reduced.spv temp.wgsl`
Function [1] '' is invalid: 
	The `break` is used outside of a `loop` or `switch` context
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', cli/src/main.rs:317:70
stack backtrace:
   0: rust_begin_unwind
             at /rustc/24bdc6d73a75dce9a7013ebc7c037013ff4ea099/library/std/src/panicking.rs:515:5
   1: core::panicking::panic_fmt
             at /rustc/24bdc6d73a75dce9a7013ebc7c037013ff4ea099/library/core/src/panicking.rs:92:14
   2: core::panicking::panic
             at /rustc/24bdc6d73a75dce9a7013ebc7c037013ff4ea099/library/core/src/panicking.rs:50:5
   3: core::option::Option<T>::unwrap
             at /rustc/24bdc6d73a75dce9a7013ebc7c037013ff4ea099/library/core/src/option.rs:388:21
   4: naga::main
             at ./cli/src/main.rs:317:56
   5: core::ops::function::FnOnce::call_once
             at /rustc/24bdc6d73a75dce9a7013ebc7c037013ff4ea099/library/core/src/ops/function.rs:227:5
```

I believe the SPIR-V is valid, and spirv-val accepts it. The "break" edge, from 1619 to 1729, is inside a switch construct - namely the construct headed at 1538. It's also inside a selection construct - the construct at 1608 - but that's allowed.

Here is the CFG:

![image](https://user-images.githubusercontent.com/4739677/123480301-11141080-d5fa-11eb-9d97-3b558c19e8bf.png)

Tint accepts this and gives this WGSL:

```
[[stage(fragment)]]
fn main() {
  switch(0u) {
    default: {
      if (false) {
        break;
      }
    }
  }
  return;
}
```

**Comment by kvark** [comment]
Thank you for filing! cc @MatusT

**Comment by jimblandy** [comment]
#940 is probably relevant to this.

**Comment by jimblandy** [comment]
With #1041, Naga accepts this and produces the WGSL:

```
fn function() {
    switch(i32(0u)) {
        default: {
            if (false) {
                break;
            } else {
                break;
            }
        }
    }
    return;
}

[[stage(fragment)]]
fn main() {
    function();
}
```

**Comment by JCapucho** [comment]
Fixed by #1294
